domain/machine/state/lxdprofile.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+
+	"github.com/juju/juju/domain/machine/internal"
+	"github.com/juju/juju/internal/errors"
+)
+
+// GetLXDProfilesForMachine returns lxd profile names mapped to the
+// lxd profile they represent for applications on the given machine.
+func (st *State) GetLXDProfilesForMachine(ctx context.Context, mName string) ([]internal.CreateLXDProfileDetails, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Capture(err)
+	}
+
+	lxdProfileStmt, err := st.Prepare(`
+SELECT (c.lxd_profile, c.revision, a.name) AS (&lxdProfileAndName.*)
+FROM   charm AS c
+JOIN   application AS a ON c.uuid = a.charm_uuid
+JOIN   unit AS u ON a.uuid = u.application_uuid
+--JOIN   net_node AS n ON u.net_node_uuid = n.uuid
```

Comment:
Should this comment be deleted?

Reply:
Yeah it's not actually required, but is nice for totality. We could just remove it.